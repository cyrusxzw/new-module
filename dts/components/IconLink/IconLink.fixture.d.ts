export namespace IconLinkFixture {
  const items: {
    altText: string;
    icon: {
      altText: string;
      large: string;
      medium: string;
      small: string;
    };
    target: string;
  }[];
}
