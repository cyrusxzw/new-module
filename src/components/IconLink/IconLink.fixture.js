const IconLinkFixture = {
  items: [
    {
      altText: 'Instagram',
      icon: {
        altText: 'instagram',
        sizes: {
          large: './assets/images/IconLink/instagram.png',
          medium: './assets/images/IconLink/instagram.png',
          small: './assets/images/IconLink/instagram.png',
        },
      },
      target: 'https://about.instagram.com/',
    },
    {
      altText: 'Twitter',
      icon: {
        altText: 'Wwitter',
        sizes: {
          large: './assets/images/IconLink/twitter.png',
          medium: './assets/images/IconLink/twitter.png',
          small: './assets/images/IconLink/twitter.png',
        },
      },
      target: 'https://twitter.com/home?lang=en',
    },
    {
      altText: 'LinkedIn',
      icon: {
        altText: 'linkedin',
        sizes: {
          large: './assets/images/IconLink/linkedin.png',
          medium: './assets/images/IconLink/linkedin.png',
          small: './assets/images/IconLink/linkedin.png',
        },
      },
      target: 'https://www.linkedin.com/home/?originalSubdomain=au',
    },
    {
      altText: 'Kakao',
      icon: {
        altText: 'kakao',
        sizes: {
          large: './assets/images/IconLink/kakao.png',
          medium: './assets/images/IconLink/kakao.png',
          small: './assets/images/IconLink/kakao.png',
        },
      },
      target: 'https://www.kakaocorp.com/service/KakaoTalk?lang=en',
    },
  ],
};

export { IconLinkFixture };
