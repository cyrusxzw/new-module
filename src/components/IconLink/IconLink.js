import React from 'react';
import PropTypes from 'prop-types';
import { Hyperlink } from '~/components/Hyperlink';
import { Image } from '~/components/Image';

const IconLink = ({
  altText,
  className,
  dataTestRef,
  hasTargetInNewWindow,
  height,
  icon,
  iconImageClassName,
  target,
  width,
}) => {
  return (
    <Hyperlink
      className={className}
      dataTestRef={dataTestRef}
      hasTargetInNewWindow={hasTargetInNewWindow}
      title={altText}
      url={target}
    >
      <Image
        altText={altText}
        className={iconImageClassName}
        sizes={icon.sizes}
        style={{ height: `${height}px`, width: `${width}px` }}
      />
    </Hyperlink>
  );
};

IconLink.propTypes = {
  altText: PropTypes.string,
  className: PropTypes.string,
  dataTestRef: PropTypes.string,
  hasTargetInNewWindow: PropTypes.bool,
  height: PropTypes.number,
  icon: PropTypes.object,
  iconImageClassName: PropTypes.string,
  target: PropTypes.string,
  width: PropTypes.number,
};

IconLink.defaultProps = {
  altText: undefined,
  className: undefined,
  dataTestRef: undefined,
  hasTargetInNewWindow: true,
  height: 21,
  icon: undefined,
  iconImageClassName: undefined,
  target: undefined,
  width: 21,
};

export { IconLink };
