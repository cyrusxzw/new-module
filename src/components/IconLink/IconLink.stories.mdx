import { Meta, Story, Canvas, ArgsTable } from '@storybook/addon-docs/blocks';
import * as knobs from '@storybook/addon-knobs';
import { Podium } from '~/components/Podium';
import { IconLink } from './IconLink';
import { IconLinkFixture } from './IconLink.fixture';

<Meta title="Components/IconLink" component={IconLink} />

# IconLink

<Canvas>
  <Story name="Base component">
    <div style={{ display: 'flex', flexWrap: 'wrap', textAlign: 'center' }}>
      {IconLinkFixture.items.map((icon, key) => (
        <div style={{ padding: '20px 40px' }} key={key}>
          <IconLink
            altText={icon.altText}
            hasTargetInNewWindow={knobs.boolean(
              'hasTargetInNewWindow',
              false,
              'Presentation',
            )}
            height={knobs.number('height', 25, 'Presentation')}
            icon={icon.icon}
            id={icon.altText}
            target={icon.target}
            width={knobs.number('width', 25, 'Presentation')}
          />
          <div style={{ marginTop: '10px' }}>{icon.altText}</div>
        </div>
      ))}
    </div>
  </Story>
</Canvas>

## API

<ArgsTable of={IconLink} />
