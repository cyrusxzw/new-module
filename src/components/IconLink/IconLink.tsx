import React from 'react';
import { Hyperlink } from '~/components/Hyperlink';
import { Image } from '~/components/Image';
import type { IconLinkType } from './IconLink.types';

const IconLink: IconLinkType = ({
  altText,
  className,
  dataTestRef,
  hasTargetInNewWindow = true,
  height = 21,
  icon,
  iconImageClassName,
  target,
  width = 21,
}) => {
  return (
    <Hyperlink
      className={className}
      dataTestRef={dataTestRef}
      hasTargetInNewWindow={hasTargetInNewWindow}
      title={altText}
      url={target}
    >
      <Image
        altText={altText}
        className={iconImageClassName}
        sizes={icon.sizes}
        style={{ height: `${height}px`, width: `${width}px` }}
      />
    </Hyperlink>
  );
};

export { IconLink };
