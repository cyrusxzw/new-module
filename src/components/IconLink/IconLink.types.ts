import type { ComponentWithoutChildren } from '~/types';
import type { ImageProps } from '../Image/Image.types';

type IconLinkProps = {
  altText?: string;
  className?: string;
  dataTestRef?: string;
  hasTargetInNewWindow?: boolean;
  height?: number;
  icon?: {
    sizes: ImageProps['sizes'];
  };
  iconImageClassName?: string;
  target?: string;
  width?: number;
};

export type IconLinkType = ComponentWithoutChildren<IconLinkProps>;
