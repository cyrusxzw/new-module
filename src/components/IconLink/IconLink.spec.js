import React from 'react';
import { render } from '@testing-library/react';
import { IconLink } from './IconLink';
import { IconLinkFixture } from './IconLink.fixture';

describe('<IconLink />', () => {
  it('should render base component correctly', () => {
    const { container } = render(
      <IconLink
        altText={IconLinkFixture.items[0].altText}
        dataTestRef="test-data-ref"
        hasTargetInNewWindow={true}
        height={25}
        icon={IconLinkFixture.items[0].icon}
        id={IconLinkFixture.items[0].altText}
        target={IconLinkFixture.items[0].target}
        width={25}
      />,
    );

    expect(container).toMatchSnapshot();
  });
});
