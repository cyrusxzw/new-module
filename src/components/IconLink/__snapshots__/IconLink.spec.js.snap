// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IconLink /> should render base component correctly 1`] = `
<div>
  <a
    class="base left dark"
    data-test-ref="test-data-ref"
    href="https://about.instagram.com/"
    target="_blank"
    title="Instagram"
  >
    <picture
      class="base picture dark"
      style="height: 25px; width: 25px;"
    >
      <source
        media="(min-width: 1025px)"
        srcset="./assets/images/IconLink/instagram.png"
      />
      <source
        media="(min-width: 640px)"
        srcset="./assets/images/IconLink/instagram.png"
      />
      <source
        media="(min-width: 0px)"
        srcset="./assets/images/IconLink/instagram.png"
      />
      <img
        alt="Instagram"
        loading="lazy"
        src="./assets/images/IconLink/instagram.png"
      />
    </picture>
  </a>
</div>
`;
